csp/middleware.py

```diff
@@ -17,6 +19,15 @@
     from django.http import HttpRequest, HttpResponseBase
 
 
+@dataclass
+class PolicyParts:
+    # A dataclass is used rather than a namedtuple so that the attributes are mutable
+    config: dict[str, Any] | None = None
+    update: dict[str, Any] | None = None
+    replace: dict[str, Any] | None = None
```

Comment:
Not sure if you can use the `_DIRECTIVES` defined in `csp.utils` here in place of the `dict[str, Any]`?

Reply:
Yep, I can do that, I avoided it because it was in a different module and the naming implied it was private to that module (that was my read at least)

Reply:
I'm open to renaming that to remove the leading `_` that implies it's private. `DIRECTIVES_T` maybe or whatever makes sense to you.

Reply:
As it's a type alias, I'd default to using camel case: `Directive`. I'll make that change if you agree

Reply:
We also have examples in `csp.decorators` for `_VIEW_T` etc. I agree that using camel case is more consistent with the `typing` names.

Reply:
I've gone with `DIRECTIVES_T`, since I was hesitant to introduce a naming style that was inconsistent with the other examples and didn't want to expand the scope of the PR to update them all.